### PR TITLE
Dev 42060  improve the initSync and the branch of the lm sdk

### PR DESCRIPTION
**Pull request by JeremyTangCD**

### CODE REVIEW

File: pkg/device/device.go [hunk 1]
```diff
@@ -49,6 +49,55 @@ func buildDevice(c *config.Config, client api.CollectorSetControllerClient, opti
 	return device
 }
 
+// checkAndUpdateExistDevice tries to find and update the devices which needs to be changed
+func (m *Manager) checkAndUpdateExistDevice(device *lm.RestDevice) (*lm.RestDevice, error) {
```
Comment (woz5999) [inline pkg/device/device.go]:
```suggestion
func (m *Manager) checkAndUpdateExistingDevice(device *lm.RestDevice) (*lm.RestDevice, error) {
```

File: pkg/device/device.go [hunk 2]
```diff
@@ -71,6 +120,15 @@ func (m *Manager) Add(options ...types.DeviceOption) (*lm.RestDevice, error) {
 
 	restResponse, apiResponse, err := m.LMClient.AddDevice(*device, false)
 	if _err := utilities.CheckAllErrors(restResponse, apiResponse, err); _err != nil {
+		if restResponse != nil && restResponse.Status == 600 {
+			log.Infof("Check and Update the existing device: %s", device.DisplayName)
+			newDevice, err := m.checkAndUpdateExistDevice(device)
```
Comment (woz5999) [inline pkg/device/device.go]:
```suggestion
			newDevice, err := m.checkAndUpdateExistingDevice(device)
```

**Comment by woz5999** [comment]
@JeremyTangCD Please implement the updated function name in my comments and then we can merge this.

**Comment by JeremyTangCD** [comment]
@woz5999 I have improved the code. Please help to check again. Thx

**Pull request merged**